lightning/src/util/ser_macros.rs

```diff
@@ -29,6 +29,9 @@ macro_rules! _encode_tlv {
 		BigSize($field.serialized_length() as u64).write($stream)?;
 		$field.write($stream)?;
 	};
+	($stream: expr, $type: expr, $field: expr, required_vec) => {
```

Comment:
I feel like the `required_` and `optional_vec` macro variants could use some shortcut docs. Basically
* if you don't want to break compat at all, use either variant with an odd TLV on write and `optional_vec` on read
* if you don't want to break compat unless something is written, use `optional_vec` with an even TLV on read/write
* if you definitely want to break compat, use `required_vec` with an even TLV on write and `optional_vec` on read